web/src/main/java/se/inera/intyg/webcert/web/web/controller/integration/IntygIntegrationController.java

```diff
@@ -266,19 +274,25 @@ public Response postRedirectToIntyg(@Context UriInfo uriInfo,
         @FormParam(PARAM_REFERENCE) String reference,
         @DefaultValue("false") @FormParam(PARAM_INACTIVE_UNIT) boolean inactiveUnit,
         @DefaultValue("false") @FormParam(PARAM_PATIENT_DECEASED) boolean deceased,
-        @DefaultValue("true") @FormParam(PARAM_FORNYA_OK) boolean fornyaOk) {
+        @DefaultValue("true") @FormParam(PARAM_FORNYA_OK) boolean fornyaOk,
+        @DefaultValue("") @FormParam(PARAM_LAUNCH_ID) String launchId) {
 
         final Map<String, Object> params = ImmutableMap.of(PARAM_CERT_ID, intygId);
 
         validateRequest(params);
 
         IntegrationParameters integrationParameters = IntegrationParameters.of(
             reference, responsibleHospName, alternatePatientSSn, fornamn, mellannamn, efternamn,
-            postadress, postnummer, postort, coherentJournaling, deceased, inactiveUnit, fornyaOk);
+            postadress, postnummer, postort, coherentJournaling, deceased, inactiveUnit, fornyaOk,
+            assertLaunchIdFormat(launchId));
 
         WebCertUser user = getWebCertUser();
         user.setParameters(integrationParameters);
 
+        if (user.getParameters().getLaunchId() != null) {
```

Comment:
Good if you add a info-log to what launchId has been added to the session. This will help us to analyze what is happening.